src/Support/SoapClients/DecoratedClient.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RicorocksDigitalAgency\Soap\Support\SoapClients;
+
+use RicorocksDigitalAgency\Soap\Contracts\PhpSoap\Client;
+use SoapClient;
+use SoapHeader;
+
+/**
+ * @internal
+ */
+final class DecoratedClient implements Client
```

Comment:
This name is a bit strange to me : sure, it decorates a built-in SoapClient.
Maybe better to name it after what it decorates?